src/python/pants/backend/python/target_types.py

```diff
@@ -724,6 +725,93 @@ def compute_value(cls, raw_value: Optional[PythonArtifact], address: Address) ->
         return cast(PythonArtifact, super().compute_value(raw_value, address))
 
 
+class PythonDistributionEntryPoints(NestedDictStringToStringField, AsyncFieldMixin):
+    alias = "entry_points"
+    required = False
+    help = (
+        "Any entry points, such as `console_scripts` and `gui_scripts`.\n"
+        "\n"
+        "Specify as a nested dictionary, with a dictionary for each type of entry point, "
+        "e.g. `console_scripts` vs. `gui_scripts`. Each dictionary maps the entry point name to "
+        'either a setuptools entry point ("path.to.module:func") or a Pants target address to a '
+        "pex_binary target.\n\n"
+        + dedent(
+            """\
+            Example:
+
+                entry_points={
+                  "console_scripts": {
+                    "my-script": "project.app:main",
+                    "another-script": "project/subdir:pex_binary_tgt"
+                  }
+                }
+
+            """
+        )
+        + "Note that Pants will assume that any value that either starts with `:` or has `/` in it, "
+        "is a target address to a pex_binary target. Otherwise, it will assume it's a setuptools "
+        "entry point as defined by "
+        "https://packaging.python.org/specifications/entry-points/#entry-points-specification. Use "
+        "`//` as a prefix for target addresses if you need to disambiguate.\n\n"
+        + dedent(
+            """\
+            Pants will attempt to infer dependencies, which you can confirm by running:
+
+                ./pants dependencies <python_distribution target address>
+
+            """
+        )
+    )
+
+
+@dataclass(frozen=True)
+class PythonDistributionPexableEntryPoint:
+    """Pexable means that the entry point could be from a pex_binary target."""
```

Comment:
There's no other type of `PythonDistribution`, so imo you can remove `Pexable`.
```suggestion
class PythonDistributionEntryPoint:
    """Note that this stores if the entry point comes from an address to a `pex_binary` target."""
```

Reply:
If you make this rename, you can rename `PythonDistributionEntryPoints` (plural) to `PythonDistributionEntryPointsField` maybe.

Reply:
Yeah, I avoided PythonDistributionEntryPoint due to there being a plural form, so a mere `s` to distinguish them by is too little imo. Also, I felt that the name `PythonDistributionEntryPointsField` got unwieldy long.. and also didn't quite follow the naming pattern for the other fields.. that doesn't have the `Field` suffix. Maybe we should start using `Field` suffix for all fields, to be consistent.. ? (it helps making it clearer when the type shows up else where, that it is a field it refers to.. which isn't always obvious, in my eyes at least)

Reply:
> Maybe we should start using Field suffix for all fields, to be consistent.. ?

Probably. We do use `Field` in a couple names if you ripgrep it, but you're right that not many yet.

Reply:
But if you prefer that, I'm happy to change it...